apps/crud/src/app/shared/item.component.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { of } from 'rxjs';
+import { ItemComponent } from './item.component';
+import { TodoConfig } from '../core/Interface/todo';
+import { By } from '@angular/platform-browser';
+
+const mockData: TodoConfig = {
+  userId: 1,
+  id: 1,
+  title: 'delectus aut autem',
+  completed: false,
+};
+
+describe('ItemComponent - ', () => {
+  let component: ItemComponent;
+  let fixture: ComponentFixture<ItemComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [ItemComponent],
+    }).compileComponents();
+  });
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ItemComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  describe('Component input and out testing', () => {
+    it('should set config and spinner state ', () => {
+      component.config = mockData;
+      component.spinnerState = of(true);
+      fixture.detectChanges();
+      expect(component.todo).toEqual(mockData);
+      component.isShowSpinner.subscribe((data) => {
+        expect(data).toEqual(true);
+      });
```

Comment:
did you try to make your test fail? Because it will never fail. The assertion is inside a subscribe, the test ends before the assertion is made. Be careful with observables

Reply:
removed the code, now using the jest to spy on the method.